### PR TITLE
Support submitOnChange option in fileTree widget

### PR DESCRIPTION
**Pull request by AlexejKossmann**
Submit after picking a file, if the submitOnChange evaluation is set to true

### CODE REVIEW

**Review by aschempp — CHANGES_REQUESTED** [review]
Why do you need the data attribute at all? If the option is set, we can just _echo_ the necessary script (`Backend.autoSubmit`)?

**Comment by AlexejKossmann** [comment]
Yes sure you are right, modified the code. Thanks

**Review by aschempp — APPROVED** [review]

**Comment by aschempp** [comment]
thanks a lot!

**Comment by leofeyer** [comment]
I'm not sure about the use-case here. @AlexejKossmann Can you elaborate on why you would want to auto-submit the form after picking a file? Usually we use auto-submit if the palette (the fields included in the form) changes depending on the value of a certain field. I fail to see how that applies to picking a file. 🙈

**Comment by AlexejKossmann** [comment]
My use-case for this modification is for picking a file of type json or csv from Contao fileSystem. The palette also contains a textarea to show a preview of the data contained by the picked file. So it is necessary to save on fileTree changes to use the load_callback of the textarea and display the data from the file. 
For me it seems better to implement contao standard behaviour, just like on a selectfield.

File: core-bundle/src/Resources/contao/widgets/FileTree.php [hunk 1]
```diff
@@ -386,6 +386,7 @@ public function generate()
                 $("ctrl_' . $this->strId . '").getParent("div").set("html", json.content);
                 json.javascript && Browser.exec(json.javascript);
                 $("ctrl_' . $this->strId . '").fireEvent("change");
+                ' . ($this->arrConfiguration['submitOnChange'] ? 'Backend.autoSubmit("' . $this->strTable . '");' : '') . '
```
Comment (leofeyer) [inline core-bundle/src/Resources/contao/widgets/FileTree.php:389]:
```suggestion
                $("ctrl_' . $this->strId . '").fireEvent("change");' . ($this->arrConfiguration['submitOnChange'] ? '
                Backend.autoSubmit("' . $this->strTable . '");' : '') . '
```

**Review by leofeyer — CHANGES_REQUESTED** [review]

File: core-bundle/src/Resources/contao/widgets/FileTree.php (hunk 1 above)
Comment (leofeyer) [reply core-bundle/src/Resources/contao/widgets/FileTree.php:389]: @AlexejKossmann It seems you have resolved this conversation without committing my suggestion or commenting why you do not want to commit it. 😄

Comment (AlexejKossmann) [reply core-bundle/src/Resources/contao/widgets/FileTree.php:389]: @leofeyer Sorry, i have overseen this change request. Too bad, i have already deleted this repository. 
Your changes are alright. Can you merge them without me, or do i need to check this commit out again?

**Comment by leofeyer** [comment]
After all, this change seems wrong to me. The `Widget` class adds the `Backend.autoSubmit()` call automatically as `onclick` or `onchange` attribute:

https://github.com/contao/contao/blob/b730ac07ddb14fde79c2bbcbdb64bedb0036c91b/core-bundle/src/Resources/contao/library/Contao/Widget.php#L1257-L1264

And the `FileTree` widget already triggers the `change` event:

https://github.com/contao/contao/blob/b730ac07ddb14fde79c2bbcbdb64bedb0036c91b/core-bundle/src/Resources/contao/widgets/FileTree.php#L374

The form should therefore be auto-submitted already. Or am I missing something?

**Comment by AlexejKossmann** [comment]
@leofeyer The FileTreeWidget _should_ have the onclick and onchange attributes from Widget, but as i have tested its not working on fileTreeWidget, the onclick is used to show the popover file-manager window. and after submitting a new file the onchange is only used to show the new filename above the submit-button. The whole backend-form where this fileTreeWidget(that has the eval: 'submitOnChange=> true') is not  submitted, and the page is not reloaded. Im not sure if it has something to do with the preventDefault https://github.com/contao/contao/blob/b730ac07ddb14fde79c2bbcbdb64bedb0036c91b/core-bundle/src/Resources/contao/widgets/FileTree.php#L362-L366

**Comment by aschempp** [comment]
@leofeyer was this intentionally closed?

**Comment by leofeyer** [comment]
No. It seems it was closed automatically when we renamed the `master` branch, because the source repo no longer exists (see https://github.com/contao/contao/pull/1436#discussion_r482356584). I think that is also why I cannot reopen the PR.

**Comment by AlexejKossmann** [comment]
@leofeyer If I recreate the pull request, will it be accepted? Or can I save myself the trouble?

**Comment by leofeyer** [comment]
The issue is still on my list for Contao 4.12, but creating a new PR would help of course. 😄

**Pull request closed without merging**